src/main/java/service/url_redirect/UrlRedirectInteractor.java

```diff
@@ -0,0 +1,36 @@
+package service.url_redirect;
+
+import java.awt.*;
+import java.net.URI;
+
+/**
+ * The Open URL Interactor.
+ * Opens the URL
+ */
+public class UrlRedirectInteractor implements UrlRedirectInputBoundary {
+    private final UrlRedirectOutputBoundary boundary;
+
+    public UrlRedirectInteractor(UrlRedirectOutputBoundary boundary) {
+        this.boundary = boundary;
+    }
+
+    /**
+     * Attempts to open urlInputData.getUrl in the default browser.
+     * @param urlInputData object that stores URL
+     */
+    public void openUrl(UrlInputData urlInputData) {
+        try {
+            final Desktop desktop = Desktop.getDesktop();
+            if (desktop.isSupported(Desktop.Action.BROWSE)) {
+                desktop.browse(new URI(urlInputData.getUrl()));
+            }
+            else {
+                boundary.displayError("Opening URLs is not supported on this system.");
+            }
+        }
+        catch (Exception exception) {
```

Comment:
Try to catch the specific exception that would be thrown. Is it necessary to have a catch if desktop.isSupported is being checked? Not sure, but either way Exception is a very generic error and it would be best to catch as specific as possible.